### PR TITLE
hypothesis: 1.14.0 -> 3.1.0

### PR DESCRIPTION
**Pull request by DamienCassou**
###### Things done:
- [x] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- [x] Built on platform(s): NixOS
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

nox-review is still not finished, I fixed everything that got reported until now.

Question: should I continue to fix every problem found by nox-review in the process of updating hypothesis, or should I better create an hypothesis3 derivation? I'm ok with both.

/cc @domenkozar @vcunat @edolstra

### CODE REVIEW

**Comment by lancelotsix** [comment]
I think the best option would be to fix errors as much as possible (which could require some work). An alternative would be to 1) rename `hypothesis` into `hypothesis1` 2) make sure that all derivations that did depend on  `hypothesis`  now depend on `hypothesis1` and then 3) introduce `hypothesis` with the desired version.

`hypothesis` could be an alias to `hypothesis3` but I am not sure if this will be the best option on the long run (we tend to prefer one version maintained).

Considering the amount of changes, the `python-wip` (build by this hydra instance https://headcounter.org/hydra/jobset/iElectric/python-wip) could be a good first destination.

Also, could it be possible to have 1 commit for each package update (makes it easier to track down)?

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -3890,17 +3891,19 @@ in modules // {
 
 
   detox = self.buildPythonPackage rec {
-    name = "detox-0.9.3";
+    name = "detox-0.10.0";
 
     propagatedBuildInputs = with self; [ tox py eventlet ];
 
+    disabled = isPy3k;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: why disabled?

**Comment by DamienCassou** [comment]
> I think the best option would be to fix errors as much as possible

done. `nox-review wip` says everything is fine.

> Considering the amount of changes, the `python-wip` could be a good first destination.

I'm done with the changes. Do you still think it is important to go through `python-wip`?

> Also, could it be possible to have 1 commit for each package update?

sure, that was planned. Done.

**Comment by lancelotsix** [comment]
I’m running nox-review on my side, this should be enough (no need for the entire hydra job).

**Comment by FRidh** [comment]
@domenkozar this PR also updates oslo/openstack. ok?

**Comment by domenkozar** [comment]
Yeah we need to test that openstack still works (I have a PR that has a NixOS test)

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -16164,6 +16211,19 @@ in modules // {
...
+  pycryptodome = buildPythonPackage rec {
+    name = "pycryptodome-3.4";
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/p/pycryptodome/${name}.tar.gz";
+      sha256 = "0nc4ry2ah56l40hr1dilw19ksma8bx3cmn99mp5vml4jlz9zkg16";
+    };
+    meta = {
+      homepage = http://www.pycryptodome.org;
+      description = "Self-contained Python package of low-level cryptographic primitives";
+      license = licences.bsd2;
```
Comment (lancelotsix) [inline pkgs/top-level/python-packages.nix]: s/licences/licenses/

**Comment by FRidh** [comment]
@domenkozar do you think that test can be merged directly so we can test whether we can merge this?
If not, I suggest @DamienCassou splits this PR into two, and follows @lancelotsix advice of creating a `hypothesis_1` and `hypothesis_3`.

**Comment by DamienCassou** [comment]
I managed to run `nox-review wip` some days ago, but `nox-review pr` gives me errors I can't reproduce with `nix-build`.

**Comment by DamienCassou** [comment]
Ok, I merged #13876. This PR is useless now.

**Pull request closed without merging**